zmapsdk/input.py

```
@@ -1,243 +1,243 @@
+        Returns:
+            Path to the created input file
+
+        Raises:
+            ZMapInputError: If an IP is invalid or file can't be created
+        """
+        # Validate IPs
+        for ip in targets:
+            try:
+                ipaddress.ip_address(ip)
+            except ValueError as e:
+                raise ZMapInputError(f"Invalid IP address: {ip} - {e!s}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)